src/overlays/actors/ovl_En_Encount4/z_en_encount4.c

```diff
@@ -33,20 +32,171 @@ const ActorInit En_Encount4_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+static s16 D_809C46D0[] = {
+    0x4000, 0xC000, 0x4000, 0, 0xC000,
+};
+static f32 D_809C46DC[] = {
+    200.0f, 200.0f, 100.0f, 0.0f, 100.0f,
+};
+
+void EnEncount4_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnEncount4* this = THIS;
+
+    this->unk148 = ENENCOUNT4_GET_F000(thisx);
+    this->switchFlags = ENENCOUNT4_GET_SWITCH_FLAGS(thisx);
+    if (this->switchFlags == 0x7F) {
+        this->switchFlags = -1;
+    }
+    if ((this->switchFlags >= 0) && Flags_GetSwitch(play, this->switchFlags)) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->actor.flags &= ~1;
+        this->actionFunc = func_809C3FD8;
+    }
+}
+
+void EnEncount4_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_809C3FD8(EnEncount4* this, PlayState* play) {
+    Actor* actor;
+
+    this->unk14E = 0;
+    if ((this->switchFlags >= 0) && Flags_GetSwitch(play, this->switchFlags)) {
+        this->timer = 100;
+        this->actionFunc = func_809C464C;
+    } else {
+        actor = play->actorCtx.actorLists[ACTORCAT_BOSS].first;
+        while (actor != NULL) {
+            if (actor->id != ACTOR_EN_BSB) {
+                actor = actor->next;
+            } else {
+                this->captainKeeta = (EnBsb*)actor;
+                this->actionFunc = func_809C4078;
+                break;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount4/EnEncount4_Init.s")
+void func_809C4078(EnEncount4* this, PlayState* play) {
+    s16 rotY;
+    Vec3f pos;
+    s32 fireWallParams;
+    s32 i;
+    EnBsb* captainKeeta = this->captainKeeta;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount4/EnEncount4_Destroy.s")
+    if ((this->switchFlags >= 0) && Flags_GetSwitch(play, this->switchFlags)) {
+        this->timer = 100;
+        this->actionFunc = func_809C464C;
+    } else if (BREG(1) == 0) {
+        if ((this->captainKeeta->actor.id != ACTOR_EN_BSB) || (captainKeeta->actor.update == NULL)) {
+            Actor_MarkForDeath(&this->actor);
+        } else if ((this->unk148 != 0) || (this->actor.xzDistToPlayer < 240.0f)) {
+            if ((this->unk148 == 0) && (captainKeeta->unk2DC != 0)) {
+                Actor_MarkForDeath(&this->actor);
+            } else {
+                fireWallParams = BGFIREWALL_PARAMS_0;
+                if ((this->unk148 == 0) || (captainKeeta->unk2DC != 0)) {
```

Comment:
```suggestion
                Actor_MarkForDeath(&this->actor);
                return;
            }
            
            fireWallParams = BGFIREWALL_PARAMS_0;
            if ((this->unk148 == 0) || (captainKeeta->unk2DC != 0)) {
```
You can put a return here and drop the else

Reply:
MarkForDeath usually gets a return.